src/libmongoc/tests/unified/operation.c

```diff
@@ -269,21 +307,36 @@ operation_list_collection_names (test_t *test,
    mongoc_cursor_t *cursor = NULL;
    char **op_ret = NULL;
    bson_error_t op_error = {0};
+   bson_parser_t *parser = NULL;
+   bson_t *opts = NULL;
+
+   parser = bson_parser_new ();
+   bson_parser_allow_extra (parser, true);
+
+   opts = bson_copy (bson_parser_get_extra (parser));
```

Comment:
Same comment about `opts = bson_new()`.

Reply:
Simplified to `opts = bson_new()` and removed `allow_extra` call.